t/unit/backends/test_mongodb.py

```diff
@@ -45,7 +45,6 @@ def setup(self):
         self.patching('celery.backends.mongodb.MongoBackend.encode')
         self.patching('celery.backends.mongodb.MongoBackend.decode')
         self.patching('celery.backends.mongodb.Binary')
-        self.patching('datetime.datetime')
```

Comment:
Why did you remove this?

Reply:
Removing that line fixes this error when running the MongoDB backend tests.
![image](https://user-images.githubusercontent.com/1480930/108774218-61d10b00-752d-11eb-9301-654b144036dc.png)

Reply:
Why does it pass on CPython?

Reply:
`tox.ini` appears to need a change to run these tests.
```diff
diff --git a/tox.ini b/tox.ini
index f62ea3cdf..51cf5d020 100644
--- a/tox.ini
+++ b/tox.ini
@@ -31,7 +31,7 @@ deps=
     3.6,3.7,3.8,3.9: -r{toxinidir}/requirements/test-ci-default.txt
     3.5,3.6,3.7,3.8,3.9: -r{toxinidir}/requirements/docs.txt
     3.6,3.7,3.8,3.9: -r{toxinidir}/requirements/docs.txt
-    pypy3: -r{toxinidir}/requirements/test-ci-base.txt
+    pypy3: -r{toxinidir}/requirements/test-ci-default.txt
 
     integration: -r{toxinidir}/requirements/test-integration.txt
 
```

Reply:
This patch was originally added back in the depths of time: 3ea8824d9b It seems to be a vestigial thing, so I think it'll be fine to remove it.

The breakage is because pypy3 appears to be a little less duck-type-y with its datetime subtraction implementation. We're giving it a `Mock` that looks like a `datetime` object, but isn't probably isn't using a `spec=` to pass `isinstance()` checks.

Reply:
Leaving this for @thedrow to resolve